### ISSUE TITLE
Can this exporter be simplified by using sudo

### ISSUE THREAD

**Issue by altmind** [cloudandheat/prometheus_smart_exporter#10]
Thanks for a great exporter.
I see that the only reason this app is split in two subcomponents is to avoid running the whole thing as root. It seems that the only place that actually need root is running smartctl subprocess.

Dont you think exporter can be dramatically simplified if there is option to skip `smart_exporter_helper` and run `sudo smartctl` directly instead. sudo is flexible enough to allow only smartctl for designated user.

**Comment by altmind** [comment]
Worth noting, that by skipping helper we can skip the dependency on pkg-config and python-systemd.

**Comment by horazont** [comment]
@altmind Yes, sudo is flexible enough. It is a trade-off between system configuration (which we cannot really manage) and dependencies (which we have control over).

In addition, this separation allows to put many more restrictions into place. Think about ProtectSystem=full, PrivateDevices (via systemd) etc. etc. all of which wouldn’t be possible if smartctl needs to run as a child of the web server process.
